src/structures/TextInputComponent.js

```diff
@@ -0,0 +1,201 @@
+   * @param {string} customId A unique string to be sent in the interaction when submitted
+   * @returns {TextInputComponent}
+   */
+  setCustomId(customId) {
+    this.customId = Util.verifyString(customId, RangeError, 'TEXT_INPUT_CUSTOM_ID');
+    return this;
+  }
+
+  /**
+   * Sets the label of this text input component
+   * @param {string} label The text to be displayed above this text input component
+   * @returns {TextInputComponent}
+   */
+  setLabel(label) {
+    this.label = Util.verifyString(label, RangeError, 'TEXT_INPUT_LABEL');
+    return this;
+  }
+
+  /**
+   * Sets the text input component to be required for modal submission
+   * @param {boolean} [required=true] Whether this text input component is required
+   * @returns {TextInputComponent}
+   */
+  setRequired(required = true) {
+    this.required = required;
+    return this;
+  }
+
+  /**
+   * Sets the maximum length of text input required in this text input component
+   * @param {number} maxLength Maximum length of text to be required
+   * @returns {TextInputComponent}
+   */
+  setMaxLength(maxLength) {
+    this.maxLength = maxLength;
+    return this;
+  }
+
+  /**
+   * Sets the minimum length of text input required in this text input component
+   * @param {number} minLength Minimum length of text to be required
+   * @returns {TextInputComponent}
+   */
+  setMinLength(minLength) {
+    this.minLength = minLength;
+    return this;
+  }
+
+  /**
+   * Sets the placeholder of this text input component
+   * @param {string} placeholder Custom placeholder text to display when no text is entered
+   * @returns {TextInputComponent}
+   */
+  setPlaceholder(placeholder) {
+    this.placeholder = Util.verifyString(placeholder, RangeError, 'TEXT_INPUT_PLACEHOLDER');
+    return this;
+  }
+
+  /**
+   * Sets the style of this text input component
+   * @param {TextInputStyleResolvable} style The style of this text input component
+   * @returns {TextInputComponent}
+   */
+  setStyle(style) {
+    this.style = TextInputComponent.resolveStyle(style);
+    return this;
+  }
+
+  /**
+   * Sets the value of this text input component
+   * @param {string} value Value of this text input component
+   * @returns {TextInputComponent}
+   */
+  setValue(value) {
+    this.value = Util.verifyString(value, RangeError, 'TEXT_INPUT_VALUE');
+    return this;
+  }
+
+  /**
+   * Transforms the text input component into a plain object
+   * @returns {APITextInput} The raw data of this text input component
+   */
+  toJSON() {
+    return {
+      custom_id: this.customId,
+      label: this.label,
+      max_length: this.maxLength,
+      min_length: this.minLength,
+      placeholder: this.placeholder,
+      required: this.required,
+      style: TextInputStyles[this.style],
+      type: MessageComponentTypes[this.type],
+      value: this.value,
+    };
+  }
+
+  /**
+   * Data that can be resolved to a TextInputStyle. This can be
+   * * TextInputStyle
+   * * number
```

Comment:
```suggestion
   * * A number
```

Reply:
Other lists `MessageComponentTypeResolvable` and `MessageButtonStyleResolvable` use `* number`, while `DateResolvable` is close with `* A number representing a timestamp`. 

I'd argue this is currently consistent as is. If changed we should change the others too.